### ISSUE TITLE
flyctl deploy -i not recognised

### ISSUE THREAD

**Issue by MattTennison** [superfly/flyctl#698]
**Describe the bug**
I have a GitHub action which calls `flyctl deploy` with a pre-build docker image. I pass the argument in using `-i` as per the [docs](https://fly.io/docs/flyctl/deploy/) and it's started erroring, complaining the shorthand isn't recognised.

It appears to have broken in the last two days based on my CI jobs:

[Successful deployment two days ago](https://github.com/MattTennison/remix-todo-app/runs/4661254389?check_suite_focus=true)
[Failed deployment today](https://github.com/MattTennison/remix-todo-app/runs/4674064602?check_suite_focus=true)
[Subsequent successful deployment with longform argument (--image)](https://github.com/MattTennison/remix-todo-app/runs/4674087549?check_suite_focus=true
)

** Add `fly.toml`

```toml
# fly.toml file generated for matttennison-todos on 2021-12-21T12:42:14Z

app = "matttennison-todos"

kill_signal = "SIGINT"
kill_timeout = 5
processes = []

[env]
  DATABASE_URL = "file:/data/sqlite.db"
  PORT = "8080"
  SECURE_COOKIE = "true"
  COOKIE_DOMAIN = "todos.matttennison.com"
  PUBLIC_ROUTE = "https://todos.matttennison.com"
  AUTH0_CALLBACK_URL = "https://todos.matttennison.com/sessions/new/callback"

[experimental]
  cmd = "start_with_migrations.sh"
  entrypoint = "sh"

[[mounts]]
  destination = "/data"
  source = "data"

[[services]]
  http_checks = []
  internal_port = 8080
  processes = ["app"]
  protocol = "tcp"
  script_checks = []

  [services.concurrency]
    hard_limit = 25
    soft_limit = 20
    type = "connections"

  [[services.ports]]
    handlers = ["http"]
    port = 80

  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443

  [[services.tcp_checks]]
    grace_period = "1s"
    interval = "15s"
    restart_limit = 0
    timeout = "2s"

```

** Command output: **

```text
Error: unknown shorthand flag: 'i' in -i
Usage:
Error unknown shorthand flag: 'i' in -i
  flyctl deploy [WORKING_DIRECTORY] [flags]



Flags:
  -a, --app string            Application name
      --build-arg strings     Set of build time variables in the form of NAME=VALUE pairs. Can be specified multiple times.
      --build-only            Build but do not deploy
      --build-target string   Set the target build stage to build if the Dockerfile has more than one stage
  -c, --config string         Path to application configuration file
      --detach                Return immediately instead of monitoring deployment progress
      --dockerfile string     Path to a Dockerfile. Defaults to the Dockerfile in the working directory.
  -e, --env strings           Set of environment variables in the form of NAME=VALUE pairs. Can be specified multiple times.
  -h, --help                  help for deploy
      --image string          The image tag or ID to deploy
      --image-label string    Image label to use when tagging and pushing to the fly registry. Defaults to "deployment-{timestamp}".
      --local-only            Only perform builds locally using the local docker daemon
      --no-cache              Do not use the build cache when building the image
      --now                   Deploy now without confirmation
  -r, --region string         The region to operate on
      --remote-only           Perform builds on a remote builder instance instead of using the local docker daemon
      --strategy string       The strategy for replacing running instances. Options are canary, rolling, bluegreen, or immediate. Default is canary, or rolling when max-per-region is set.

Global Flags:
  -t, --access-token string   Fly API Access Token
  -j, --json                  json output
      --verbose               verbose output
```

**Comment by azazeal** [comment]
Hi there - a fix for this will get out shortly. Thank you for reporting it.

**Comment by jsierles** [comment]
Fixed in #699

**Comment by MattTennison** [comment]
Thanks for getting this resolved so quickly :smile: 

I'll keep an eye out for the next release, and re-run the failed workflow.
